util/cache/cache.go

```diff
@@ -129,6 +139,61 @@ func getFlagVal[T any](cmd *cobra.Command, o Options, name string, getVal func(n
 	}
 }
 
+// loadRedisCredsFromFile loads Redis credentials from a file mounted at the specified path.
+func loadRedisCredsFromSecret(mountPath string, opt Options) *redisCreds {
+	creds := &redisCreds{}
+	if mountPath != "" {
+		log.Info("Loading Redis credentials from file: ", mountPath)
+		readAuthDetailsFromFile := func(filename string) string {
+			data, err := os.ReadFile(filepath.Join(mountPath, filename))
+			if err != nil {
+				return ""
```

Comment:
```suggestion
				return err, ""
```
let's handle error here.